app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt

```diff
@@ -73,10 +73,19 @@ class EditPatientEffectHandler @AssistedInject constructor(
         .addTransformer(FetchBpPassportsEffect::class.java, fetchBpPassports(schedulersProvider.io()))
         .addTransformer(SavePatientEffect::class.java, savePatientTransformer(schedulersProvider.io()))
         .addTransformer(LoadInputFields::class.java, loadInputFields())
-        .addConsumer(SetupUi::class.java, { ui.setupUi(it.inputFields)}, schedulersProvider.ui())
+        .addConsumer(SetupUi::class.java, { ui.setupUi(it.inputFields) }, schedulersProvider.ui())
+        .addTransformer(FetchColonyOrVillagesEffect::class.java, fetchColonyOrVillages())
         .build()
   }
 
+  private fun fetchColonyOrVillages(): ObservableTransformer<FetchColonyOrVillagesEffect, EditPatientEvent> {
+    var colonyOrVillages: List<String>? = null
+    return ObservableTransformer { fetchColonyOrVillagesEffect ->
+      fetchColonyOrVillagesEffect.doOnNext { colonyOrVillages = patientRepository.allColoniesOrVillagesInPatientAddress() }
+          .map { ColonyOrVillagesFetched(colonyOrVillages) }
+    }
```

Comment:
We should not be using a `var` variable here to pass the list information downstream. Instead, use `map`.

```kt
fetchColonyOrVillagesEffect
    .map { patientRepository.allColoniesOrVillagesInPatientAddress() }
    .map(::ColonyOrVillagesFetched)
```